types/three/three-core.d.ts

```diff
@@ -5200,10 +5200,21 @@ export class LOD extends Object3D {
     objects: any[];
 }
 
+
+/**
+ * An intermediate material type that casts more precisely the possible materials assignable to a [[Line]] object.
+ *
+ * [[LineDashedMaterial]] is omitted as it extends [[LineBasicMaterial]].
+ *
+ * // Todo:
+ * // - can [[Line]] take in an array of materials ?
+ */
+export type LineMaterialType = LineBasicMaterial | ShaderMaterial | MeshDepthMaterial;
```

Comment:
Why is `MeshDepthMaterial` a part of this list?

Reply:
MeshDepthMaterial is applied to all the scene as an overrideMaterial in some postprocessing. I remember that I tested MeshDepthMaterial on a line and that it did render as expected. Since there is no LineDepthMaterial, this material is expected to be applied on a Line by the library.

Reply:
But it wouldn't make sense to pass `MeshDepthMaterial` into a `Line` constructor, right?

Reply:
I don't know. One could imagine a mask effect with lines, based on depth, and assigning the MeshDepthMaterial in the constructor would work. If the types admit so, then effects can also be imagined with MeshBasicMaterial, and I think that there is no technical issue with this material on a line. Then the other materials have the problems of using the normal attribute, missing in a Line geometry. So, I don't know. This is about a very special use case, and one can also use the overrideMaterial anyway. As you wish.